pkg/kubelet/cm/cpumanager/topology/topology.go

```diff
@@ -62,6 +62,30 @@ func (topo *CPUTopology) CPUsPerSocket() int {
 	return topo.NumCPUs / topo.NumSockets
 }
 
+// CPUCoreSiblings returns a CPUSet containing all the core siblings
+// of the given cpuid, including itself. Returns empty set if the cpuid is unknown.
+func (topo *CPUTopology) CPUCoreSiblings(id int) cpuset.CPUSet {
+	d, ok := topo.CPUDetails[id]
+	if !ok {
+		return cpuset.New()
+	}
+	if topo.CPUsPerCore() == 1 {
```

Comment:
I think we should move this check up (right at the beginning). If hyperthreading is not enabled in the system, we return the cpu id straightaway and don't have to obtain the cpuinfo in that case.

Reply:
right, fixing.